Integrations/integration-PagerDuty.yml

```diff
@@ -0,0 +1,475 @@
+                                   params=params_dict,
+                                   headers=DEFAULT_HEADERS,
+                                  )
+            res.raise_for_status()
+
+            return res.json()
+
+        except Exception, e:
+            LOG(e)
+            raise(e)
+
+
+    def test_module():
+        try:
+            get_users_on_call_now_command()
+        except Exception, e:
+            raise Exception(e.message)
+
+        demisto.results("ok")
+
+    def extract_on_call_user_data(users):
+        """Extact data about user from a given schedule."""
+        outputs = []
+        context = []
+        for i, user in enumerate(users):
+            output = {}
+
+            output['ID'] = user.get('id')
+            output['Name'] = user.get('name')
+            output['Role'] = user.get('role')
+            output['Color'] = user.get('color')
+            output['Email'] = user.get('email')
+            output['User Url'] = user.get('html_url')
+            output['Time Zone'] = user.get('time_zone')
+            output['Avatar Url'] = user.get('avatar_url')
+
+            outputs.append(output)
+            context.append(output)
+
+        return {
+            'Type': entryTypes['note'],
+            'Contents': outputs,
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown(USERS_ON_CALL, outputs, USERS_ON_CALL_NOW_HEADERS),
+            'EntryContext': {
+                'PagerDutyUser': context
+            }
+        }
+
+    def extract_on_call_now_user_data(oncalls, table_name):
+        """Extract the user data from the oncalls json."""
+        outputs = []
+        context = []
+        for i in xrange(len(oncalls)):
+            output = {}
+            data = oncalls[i]
+            user = data.get('user')
+
+            output['ID'] = user.get('id')
+            output['Name'] = user.get('name')
+            output['Role'] = user.get('role')
+            output['Color'] = user.get('color')
+            output['Email'] = user.get('email')
+            output['User Url'] = user.get('html_url')
+            output['Time Zone'] = user.get('time_zone')
+            output['Avatar Url'] = user.get('avatar_url')
+
+            escal_level = data.get('escalation_level', 1)
+            outputs.insert(escal_level - 1, output)
+            context.insert(escal_level - 1, output)
+
+        return {
+            'Type': entryTypes['note'],
+            'Contents': outputs,
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown(table_name, outputs, USERS_ON_CALL_NOW_HEADERS),
+            'EntryContext': {
+                'PagerDutyUser': context
+            }
+        }
+
+    def extract_incidents_data(incidents):
+        """Extact data about incidents."""
+        outputs = []
```

Comment:
@ronykoz it seems that outputs and context are identical here. If so we can leave just one of them and use it both for `Contents` and `EntryContext`

Reply:
Done